### PR TITLE
fix: handle default features being disabled

### PR DESCRIPTION
**Pull request by flavio**
When the default features are disabled, the k8s-openapi crate is no longer required.

The recent helpers introduced to deal with Kubernetes resources containing Pod didn't take that into account.

That causes the library to not compile properly when the imported without the default features.

This commit fixes this and extends our unit tests to ensure this won't happen again.

### CODE REVIEW

**Review by jvanz — APPROVED** [review]
LGTM. But it's necessary to check why the tests running in the CI failed. In my local environment I could run them with no modifications.

**Comment by flavio** [comment]
> LGTM. But it's necessary to check why the tests running in the CI failed. In my local environment I could run them with no modifications.

I made a mistake inside of the GitHub action definition. this should be fixed now, I'll wait for the run to complete and then I'll merge the PR if everything is green

**Pull request merged**
